src/PaymentGateways/PayPalCommerce/MerchantDetail.php

```diff
@@ -0,0 +1,215 @@
+<?php
+namespace Give\PaymentGateways\PayPalCommerce;
+
+use InvalidArgumentException;
+
+/**
+ * Class MerchantDetail
+ * @package Give\PaymentGateways\PayPalCommerce
+ *
+ * @since 2.8.0
+ */
+class MerchantDetail {
+	/**
+	 * PayPal merchant Id  (email address)
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var null|string
+	 */
+	public $merchantId = null;
+
+	/**
+	 * PayPal merchant id
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var null|string
+	 */
+	public $merchantIdInPayPal = null;
+
+	/**
+	 * Client id.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var null |string
+	 */
+	public $clientId = null;
+
+	/**
+	 * Client Secret
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var null|string
+	 */
+	public $clientSecret = null;
+
+	/**
+	 * Environment mode.
+	 *
+	 * @var null|string
+	 */
+	private $mode = null;
+
+	/**
+	 * Access token.
+	 *
+	 * @var null|string
+	 */
+	public $accessToken = null;
+
+	/**
+	 * Access token.
+	 *
+	 * @var array
+	 */
+	private $tokenDetails = null;
+
+	/**
+	 * MerchantDetail constructor.
+	 */
+	public function __construct() {
+		$this->mode = give( PayPalClient::class )->mode;
+	}
+
+	/**
+	 * Define properties values.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return $this
+	 */
+	public function boot() {
+		$paypalAccount = get_option( OptionId::$payPalAccountsOptionKey, [] );
+
+		if ( $paypalAccount ) {
+			$this->validate( $paypalAccount );
+			$this->setupProperties( $paypalAccount );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Save merchant details.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return bool
+	 */
+	public function save() {
+		return update_option( OptionId::$payPalAccountsOptionKey, $this->toArray() );
+	}
+
+	/**
+	 * Delete merchant details.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return bool
+	 */
+	public function delete() {
+		return delete_option( OptionId::$payPalAccountsOptionKey );
+	}
+
+	/**
+	 * Return array of merchnat details.
+	 *
+	 * @sicne 2.8.0
+	 *
+	 * @return array
+	 */
+	public function toArray() {
+		return [
+			'merchantId'         => $this->merchantId,
+			'merchantIdInPayPal' => $this->merchantIdInPayPal,
+			$this->mode          => [
+				'clientId'     => $this->clientId,
+				'clientSecret' => $this->clientSecret,
+				'token'        => $this->tokenDetails,
+			],
+		];
+	}
+
+	/**
+	 * Make MerchantDetail object from array.
+	 *
+	 * @param array $merchantDetails
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return MerchantDetail
+	 */
+	public function fromArray( $merchantDetails ) {
+		$obj = new static();
+
+		$obj->setupProperties( $merchantDetails );
```

Comment:
Make sure to validate here, as well.

Reply:
fixed: https://github.com/impress-org/givewp/compare/4eb148d59e6a...0e0e27168d4b